-1

**Failed Tests**:   Build 
**Size**: This PR adds an extra 28KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d92a11/44452/summary.html
**COMMIT**: e3658fe1474d4200a9921f0a81b4db0fdfc6156e
**CMSSW**: CMSSW_15_1_X_2025-02-17-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/47357/44452/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>>> Compiling edm plugin src/L1Trigger/L1ExtraFromDigis/src/L1ExtraParticlesProd.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -c -DCMS_MICRO_ARCH='x86-64-v3' -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DCMSSW_GIT_HASH='CMSSW_15_1_X_2025-02-17-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_15_1_X_2025-02-17-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_15_1_X_2025-02-17-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/pcre/8.43-2d141998cfe5424b8f7aff48035cc2da/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/boost/1.80.0-96a02191111b66819e07de179cb25a0e/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/clhep/2.4.7.1-d3a3e353d370e701238f7949a0d7909f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/gsl/2.6-f7574c606b0ce57ff601d3ca9534cd01/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/hls/2019.08-e6beae7d560007d8bb20c2cf88bfde9a/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/lcg/root/6.32.09-0a945d6e24dcaabe218b38fa8292785d/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/tbb/v2021.9.0-9b0f135342cfe979b6500c59f501774e/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/utm/utm_0.13.0-72ee52d69791ed49755d3e16b24e6f1e/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/xerces-c/3.1.3-c7b88eaa36d0408120f3c29826a04bf6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/zlib/1.2.13-d217cdbdd8d586e845e05946de2796be/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-e265b266d2b30c1bebdd883980d0f9d0/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-e265b266d2b30c1bebdd883980d0f9d0/include/eigen3 -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/fmt/10.2.1-e35fd1db5eb3abc8ac0452e8ee427196/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/OpenBLAS/0.3.27-70a9dd2c9f309171934f13e3003b0540/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/tinyxml2/6.2.0-f99ae2781d074227d47e8a3e7c8ec87e/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc12/src/L1Trigger/L1ExtraFromDigis/src/L1TriggerL1ExtraFromDigis/L1ExtraParticlesProd.cc.d src/L1Trigger/L1ExtraFromDigis/src/L1ExtraParticlesProd.cc -o tmp/el8_amd64_gcc12/src/L1Trigger/L1ExtraFromDigis/src/L1TriggerL1ExtraFromDigis/L1ExtraParticlesProd.cc.o
>> Compiling edm plugin src/L1Trigger/L1ExtraFromDigis/src/L1ExtraTestAnalyzer.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -c -DCMS_MICRO_ARCH='x86-64-v3' -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DCMSSW_GIT_HASH='CMSSW_15_1_X_2025-02-17-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_15_1_X_2025-02-17-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_15_1_X_2025-02-17-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/pcre/8.43-2d141998cfe5424b8f7aff48035cc2da/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/boost/1.80.0-96a02191111b66819e07de179cb25a0e/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/clhep/2.4.7.1-d3a3e353d370e701238f7949a0d7909f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/gsl/2.6-f7574c606b0ce57ff601d3ca9534cd01/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/hls/2019.08-e6beae7d560007d8bb20c2cf88bfde9a/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/lcg/root/6.32.09-0a945d6e24dcaabe218b38fa8292785d/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/tbb/v2021.9.0-9b0f135342cfe979b6500c59f501774e/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/utm/utm_0.13.0-72ee52d69791ed49755d3e16b24e6f1e/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/xerces-c/3.1.3-c7b88eaa36d0408120f3c29826a04bf6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/zlib/1.2.13-d217cdbdd8d586e845e05946de2796be/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-e265b266d2b30c1bebdd883980d0f9d0/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-e265b266d2b30c1bebdd883980d0f9d0/include/eigen3 -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/fmt/10.2.1-e35fd1db5eb3abc8ac0452e8ee427196/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/OpenBLAS/0.3.27-70a9dd2c9f309171934f13e3003b0540/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02877/el8_amd64_gcc12/external/tinyxml2/6.2.0-f99ae2781d074227d47e8a3e7c8ec87e/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc12/src/L1Trigger/L1ExtraFromDigis/src/L1TriggerL1ExtraFromDigis/L1ExtraTestAnalyzer.cc.d src/L1Trigger/L1ExtraFromDigis/src/L1ExtraTestAnalyzer.cc -o tmp/el8_amd64_gcc12/src/L1Trigger/L1ExtraFromDigis/src/L1TriggerL1ExtraFromDigis/L1ExtraTestAnalyzer.cc.o
src/L1Trigger/L1ExtraFromDigis/src/L1ExtraParticlesProd.cc: In member function 'virtual void L1ExtraParticlesProd::produce(edm::Event&, const edm::EventSetup&)':
src/L1Trigger/L1ExtraFromDigis/src/L1ExtraParticlesProd.cc:198:42: error: variable 'muHandle' set but not used [-Werror=unused-but-set-variable]
  198 |   OrphanHandle<L1MuonParticleCollection> muHandle = iEvent.put(std::move(muColl));
      |                                          ^~~~~~~~
src/L1Trigger/L1ExtraFromDigis/src/L1ExtraParticlesProd.cc:850:40: error: variable 'isoEmHandle' set but not used [-Werror=unused-but-set-variable]
  850 |   OrphanHandle<L1EmParticleCollection> isoEmHandle = iEvent.put(std::move(isoEmColl), "Isolated");
      |                                        ^~~~~~~~~~~
</pre>

--- a/CommonTools/RecoUtils/plugins/LeptonFSRProducer.cc
+++ b/CommonTools/RecoUtils/plugins/LeptonFSRProducer.cc
@@ -10,23 +10,20 @@
  */
 
 #include <memory>
-#include "FWCore/Framework/interface/Frameworkfwd.h"
-#include "FWCore/Framework/interface/global/EDProducer.h"
-
-#include "FWCore/Framework/interface/Event.h"
-#include "FWCore/Framework/interface/MakerMacros.h"
-
-#include "FWCore/ParameterSet/interface/ParameterSet.h"
-#include "FWCore/Utilities/interface/StreamID.h"
 
 #include "DataFormats/Candidate/interface/Candidate.h"
-#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
-#include "DataFormats/PatCandidates/interface/GenericParticle.h"
-#include "DataFormats/Math/interface/LorentzVector.h"
-
-#include "DataFormats/PatCandidates/interface/Muon.h"
-#include "DataFormats/PatCandidates/interface/Electron.h"
 #include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "DataFormats/PatCandidates/interface/Electron.h"
+#include "DataFormats/PatCandidates/interface/GenericParticle.h"
+#include "DataFormats/PatCandidates/interface/Muon.h"
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
 
 class LeptonFSRProducer : public edm::global::EDProducer<> {
 public:
@@ -217,7 +214,7 @@ void LeptonFSRProducer::produce(edm::StreamID streamID, edm::Event& iEvent, cons
     }
   }  // end of loop over pfCands
 
-  edm::OrphanHandle<std::vector<pat::GenericParticle>> oh = iEvent.put(std::move(fsrPhotons));
+  iEvent.put(std::move(fsrPhotons));
 
   {
     std::unique_ptr<edm::ValueMap<int>> bareIdx(new edm::ValueMap<int>());

--- a/CommonTools/UtilAlgos/interface/AssociationVectorSelector.h
+++ b/CommonTools/UtilAlgos/interface/AssociationVectorSelector.h
@@ -68,9 +68,10 @@ void AssociationVectorSelector<KeyRefProd, CVal, KeySelector, ValSelector>::prod
   KeyRefProd ref = evt.getRefBeforePut<collection_t>();
   unique_ptr<association_t> selectedAssociation(new association_t(ref, selected.get()));
   size = selected->size();
-  OrphanHandle<collection_t> oh = evt.put(std::move(selected));
-  for (size_t i = 0; i != size; ++i)
+  evt.put(std::move(selected));
+  for (size_t i = 0; i != size; ++i) {
     selectedAssociation->setValue(i, selectedValues[i]);
+  }
   evt.put(std::move(selectedAssociation));
 }
 

--- a/EventFilter/GctRawToDigi/plugins/L1GctInternJetProducer.cc
+++ b/EventFilter/GctRawToDigi/plugins/L1GctInternJetProducer.cc
@@ -14,14 +14,12 @@
 #include <memory>
 
 // user include files
-#include "EventFilter/GctRawToDigi/plugins/L1GctInternJetProducer.h"
-
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/ESHandle.h"
 #include "FWCore/Framework/interface/Frameworkfwd.h"
 #include "FWCore/Framework/interface/MakerMacros.h"
 
-#include "DataFormats/Common/interface/Handle.h"
-#include "FWCore/Framework/interface/ESHandle.h"
-#include "DataFormats/Common/interface/OrphanHandle.h"
+#include "L1GctInternJetProducer.h"
 
 L1GctInternJetProducer::L1GctInternJetProducer(const edm::ParameterSet& iConfig)
     : internalJetSource_(iConfig.getParameter<edm::InputTag>("internalJetSource")),
@@ -80,7 +78,7 @@ void L1GctInternJetProducer::produce(edm::StreamID, edm::Event& iEvent, const ed
     }
   }
 
-  OrphanHandle<L1JetParticleCollection> internalJetHandle = iEvent.put(std::move(internJetColl), "Internal");
+  iEvent.put(std::move(internJetColl), "Internal");
 }
 
 //define this as a plug-in

--- a/L1Trigger/L1ExtraFromDigis/src/L1ExtraParticlesProd.cc
+++ b/L1Trigger/L1ExtraFromDigis/src/L1ExtraParticlesProd.cc
@@ -195,28 +195,20 @@ void L1ExtraParticlesProd::produce(edm::Event &iEvent, const edm::EventSetup &iS
     }
   }
 
-  OrphanHandle<L1MuonParticleCollection> muHandle = iEvent.put(std::move(muColl));
+  iEvent.put(std::move(muColl));
 
   // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   // ~~~~~~~~~~~~~~~~~~~~ Calorimeter ~~~~~~~~~~~~~~~~~~~~
   // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
   unique_ptr<L1EmParticleCollection> isoEmColl(new L1EmParticleCollection);
-
   unique_ptr<L1EmParticleCollection> nonIsoEmColl(new L1EmParticleCollection);
-
   unique_ptr<L1JetParticleCollection> cenJetColl(new L1JetParticleCollection);
-
   unique_ptr<L1JetParticleCollection> forJetColl(new L1JetParticleCollection);
-
   unique_ptr<L1JetParticleCollection> tauJetColl(new L1JetParticleCollection);
-
   unique_ptr<L1JetParticleCollection> isoTauJetColl(new L1JetParticleCollection);
-
   unique_ptr<L1EtMissParticleCollection> etMissColl(new L1EtMissParticleCollection);
-
   unique_ptr<L1EtMissParticleCollection> htMissColl(new L1EtMissParticleCollection);
-
   unique_ptr<L1HFRingsCollection> hfRingsColl(new L1HFRingsCollection);
 
   if (produceCaloParticles_) {
@@ -847,23 +839,15 @@ void L1ExtraParticlesProd::produce(edm::Event &iEvent, const edm::EventSetup &iS
     }
   }
 
-  OrphanHandle<L1EmParticleCollection> isoEmHandle = iEvent.put(std::move(isoEmColl), "Isolated");
-
-  OrphanHandle<L1EmParticleCollection> nonIsoEmHandle = iEvent.put(std::move(nonIsoEmColl), "NonIsolated");
-
-  OrphanHandle<L1JetParticleCollection> cenJetHandle = iEvent.put(std::move(cenJetColl), "Central");
-
-  OrphanHandle<L1JetParticleCollection> forJetHandle = iEvent.put(std::move(forJetColl), "Forward");
-
-  OrphanHandle<L1JetParticleCollection> tauJetHandle = iEvent.put(std::move(tauJetColl), "Tau");
-
-  OrphanHandle<L1JetParticleCollection> IsoTauJetHandle = iEvent.put(std::move(isoTauJetColl), "IsoTau");
-
-  OrphanHandle<L1EtMissParticleCollection> etMissCollHandle = iEvent.put(std::move(etMissColl), "MET");
-
-  OrphanHandle<L1EtMissParticleCollection> htMissCollHandle = iEvent.put(std::move(htMissColl), "MHT");
-
-  OrphanHandle<L1HFRingsCollection> hfRingsCollHandle = iEvent.put(std::move(hfRingsColl));
+  iEvent.put(std::move(isoEmColl), "Isolated");
+  iEvent.put(std::move(nonIsoEmColl), "NonIsolated");
+  iEvent.put(std::move(cenJetColl), "Central");
+  iEvent.put(std::move(forJetColl), "Forward");
+  iEvent.put(std::move(tauJetColl), "Tau");
+  iEvent.put(std::move(isoTauJetColl), "IsoTau");
+  iEvent.put(std::move(etMissColl), "MET");
+  iEvent.put(std::move(htMissColl), "MHT");
+  iEvent.put(std::move(hfRingsColl));
 }
 
 // math::XYZTLorentzVector

--- a/L1Trigger/VertexFinder/plugins/TPStubValueMapProducer.cc
+++ b/L1Trigger/VertexFinder/plugins/TPStubValueMapProducer.cc
@@ -191,7 +191,7 @@ void TPStubValueMapProducer::produce(edm::StreamID, edm::Event& iEvent, const ed
 
   // Collections of products
   auto vTrackingParticlesHandle = iEvent.put(std::move(vTrackingParticles));
-  auto vAllMatchedTPsHandle = iEvent.put(std::move(vAllMatchedTPs), outputCollectionNames_[2]);
+  iEvent.put(std::move(vAllMatchedTPs), outputCollectionNames_[2]);
 
   // Value maps to TP/Stub
   auto TPV = std::make_unique<edm::ValueMap<l1tVertexFinder::TP>>();

--- a/MuonAnalysis/MuonAssociators/plugins/MuonFSRProducer.cc
+++ b/MuonAnalysis/MuonAssociators/plugins/MuonFSRProducer.cc
@@ -1,23 +1,19 @@
 // system include files
 #include <memory>
 
-// user include files
-#include "FWCore/Framework/interface/Frameworkfwd.h"
-#include "FWCore/Framework/interface/global/EDProducer.h"
-
+// CMSSW include files
+#include "DataFormats/Candidate/interface/Candidate.h"
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "DataFormats/PatCandidates/interface/Electron.h"
+#include "DataFormats/PatCandidates/interface/GenericParticle.h"
+#include "DataFormats/PatCandidates/interface/Muon.h"
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
 #include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
 #include "FWCore/Framework/interface/MakerMacros.h"
-
+#include "FWCore/Framework/interface/global/EDProducer.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "FWCore/Utilities/interface/StreamID.h"
-
-#include "DataFormats/Candidate/interface/Candidate.h"
-#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
-#include "DataFormats/PatCandidates/interface/GenericParticle.h"
-#include "DataFormats/Math/interface/LorentzVector.h"
-
-#include "DataFormats/PatCandidates/interface/Muon.h"
-#include "DataFormats/PatCandidates/interface/Electron.h"
 
 //
 // class declaration
@@ -182,7 +178,7 @@ void MuonFSRProducer::produce(edm::StreamID streamID, edm::Event& iEvent, const 
     }
   }
 
-  edm::OrphanHandle<std::vector<pat::GenericParticle>> oh = iEvent.put(std::move(fsrPhotons));
+  iEvent.put(std::move(fsrPhotons));
 }
 
 double MuonFSRProducer::computeRelativeIsolation(const pat::PackedCandidate& photon,

--- a/PhysicsTools/PatAlgos/plugins/PATGenCandsFromSimTracksProducer.cc
+++ b/PhysicsTools/PatAlgos/plugins/PATGenCandsFromSimTracksProducer.cc
@@ -9,23 +9,21 @@
   \version  $Id: PATGenCandsFromSimTracksProducer.cc,v 1.8 2010/10/20 23:09:25 wmtan Exp $
 */
 
-#include "FWCore/Framework/interface/Frameworkfwd.h"
-#include "FWCore/Framework/interface/stream/EDProducer.h"
-#include "FWCore/Framework/interface/Event.h"
-#include "FWCore/Framework/interface/MakerMacros.h"
-#include "FWCore/ParameterSet/interface/ParameterSet.h"
-
-#include "DataFormats/HepMCCandidate/interface/GenParticle.h"
-#include "DataFormats/HepMCCandidate/interface/GenParticleFwd.h"
-#include "SimDataFormats/Track/interface/SimTrackContainer.h"
-#include "SimDataFormats/Vertex/interface/SimVertexContainer.h"
-
-#include "CommonTools/Utils/interface/StringCutObjectSelector.h"
-#include "SimGeneral/HepPDTRecord/interface/PdtEntry.h"
-#include "SimGeneral/HepPDTRecord/interface/ParticleDataTable.h"
-
 #include <ext/algorithm>
 #include <memory>
+
+#include "CommonTools/Utils/interface/StringCutObjectSelector.h"
+#include "DataFormats/HepMCCandidate/interface/GenParticle.h"
+#include "DataFormats/HepMCCandidate/interface/GenParticleFwd.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "SimDataFormats/Track/interface/SimTrackContainer.h"
+#include "SimDataFormats/Vertex/interface/SimVertexContainer.h"
+#include "SimGeneral/HepPDTRecord/interface/ParticleDataTable.h"
+#include "SimGeneral/HepPDTRecord/interface/PdtEntry.h"
 
 namespace pat {
   class PATGenCandsFromSimTracksProducer : public edm::stream::EDProducer<> {
@@ -326,7 +324,10 @@ void PATGenCandsFromSimTracksProducer::produce(Event &event, const EventSetup &i
   }
 
   // Write to the Event, and get back a handle (which can be useful for debugging)
-  edm::OrphanHandle<reco::GenParticleCollection> orphans = event.put(std::move(cands));
+#ifdef DEBUG_PATGenCandsFromSimTracksProducer
+  edm::OrphanHandle<reco::GenParticleCollection> orphans =
+#endif
+      event.put(std::move(cands));
 
 #ifdef DEBUG_PATGenCandsFromSimTracksProducer
   std::cout << "Produced a list of " << orphans->size() << " genParticles." << std::endl;

--- a/RecoEcal/EgammaClusterProducers/src/SCEnergyCorrectorDRNProducer.cc
+++ b/RecoEcal/EgammaClusterProducers/src/SCEnergyCorrectorDRNProducer.cc
@@ -99,7 +99,7 @@ void SCEnergyCorrectorDRNProducer::produce(edm::Event& iEvent, const edm::EventS
     ++i;
   }
 
-  auto scHandle = iEvent.put(std::move(corrSCs));
+  iEvent.put(std::move(corrSCs));
 }
 
 void SCEnergyCorrectorDRNProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {

--- a/RecoLocalFastTime/FTLRecProducers/plugins/MTDRecHitProducer.cc
+++ b/RecoLocalFastTime/FTLRecProducers/plugins/MTDRecHitProducer.cc
@@ -106,9 +106,8 @@ void MTDRecHitProducer::produce(edm::Event& evt, const edm::EventSetup& es) {
   }
 
   // put the collection of recunstructed hits in the event
-  // get the orphan handles so we can make refs for the tracking rechits
-  auto barrelHandle = evt.put(std::move(barrelRechits), ftlbInstance_);
-  auto endcapHandle = evt.put(std::move(endcapRechits), ftleInstance_);
+  evt.put(std::move(barrelRechits), ftlbInstance_);
+  evt.put(std::move(endcapRechits), ftleInstance_);
 }
 
 #include "FWCore/Framework/interface/MakerMacros.h"

--- a/RecoMTD/TrackExtender/plugins/TrackExtenderWithMTD.cc
+++ b/RecoMTD/TrackExtender/plugins/TrackExtenderWithMTD.cc
@@ -1,64 +1,48 @@
-#include "FWCore/Framework/interface/Frameworkfwd.h"
-#include "FWCore/Framework/interface/stream/EDProducer.h"
-#include "FWCore/Framework/interface/Event.h"
-#include "FWCore/Framework/interface/EventSetup.h"
-#include "FWCore/Framework/interface/ESHandle.h"
-#include "FWCore/Framework/interface/ConsumesCollector.h"
-#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include <sstream>
 
-#include "RecoMTD/DetLayers/interface/MTDDetLayerGeometry.h"
-#include "RecoMTD/Records/interface/MTDRecoGeometryRecord.h"
-
-#include "MagneticField/Engine/interface/MagneticField.h"
-#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
-
-#include "TrackingTools/KalmanUpdators/interface/Chi2MeasurementEstimator.h"
-
-#include "DataFormats/TrackerRecHit2D/interface/MTDTrackingRecHit.h"
-
-#include "RecoMTD/DetLayers/interface/MTDTrayBarrelLayer.h"
-#include "TrackingTools/DetLayers/interface/ForwardDetLayer.h"
+#include <CLHEP/Units/GlobalPhysicalConstants.h>
 
 #include "DataFormats/ForwardDetId/interface/BTLDetId.h"
 #include "DataFormats/ForwardDetId/interface/ETLDetId.h"
 #include "DataFormats/ForwardDetId/interface/MTDChannelIdentifier.h"
-#include "Geometry/CommonTopologies/interface/PixelTopology.h"
 #include "DataFormats/GeometryVector/interface/GlobalPoint.h"
-
-#include "TrackingTools/PatternTools/interface/Trajectory.h"
-#include "TrackingTools/PatternTools/interface/TrajTrackAssociation.h"
-
-#include "TrackingTools/TransientTrack/interface/TransientTrack.h"
-#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
-#include "TrackingTools/Records/interface/TransientTrackRecord.h"
-
-#include "TrackingTools/TransientTrackingRecHit/interface/TransientTrackingRecHit.h"
-
-#include "RecoMTD/TransientTrackingRecHit/interface/MTDTransientTrackingRecHitBuilder.h"
-#include "TrackingTools/Records/interface/TransientRecHitRecord.h"
-
-#include "TrackingTools/Records/interface/TrackingComponentsRecord.h"
-#include "TrackingTools/GeomPropagators/interface/Propagator.h"
-
-#include "TrackingTools/PatternTools/interface/TSCBLBuilderWithPropagator.h"
-
-#include "RecoTracker/TransientTrackingRecHit/interface/Traj2TrackHits.h"
-#include "TrackingTools/TrackRefitter/interface/TrackTransformer.h"
-
-#include <sstream>
-
-#include "Geometry/CommonTopologies/interface/Topology.h"
-
-#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
-#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
-
 #include "DataFormats/Math/interface/GeantUnits.h"
 #include "DataFormats/Math/interface/LorentzVector.h"
-#include "CLHEP/Units/GlobalPhysicalConstants.h"
 #include "DataFormats/Math/interface/Rounding.h"
-
-#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "DataFormats/TrackerRecHit2D/interface/MTDTrackingRecHit.h"
 #include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#include "Geometry/CommonTopologies/interface/Topology.h"
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+#include "RecoMTD/DetLayers/interface/MTDDetLayerGeometry.h"
+#include "RecoMTD/DetLayers/interface/MTDTrayBarrelLayer.h"
+#include "RecoMTD/Records/interface/MTDRecoGeometryRecord.h"
+#include "RecoMTD/TransientTrackingRecHit/interface/MTDTransientTrackingRecHitBuilder.h"
+#include "RecoTracker/TransientTrackingRecHit/interface/Traj2TrackHits.h"
+#include "TrackingTools/DetLayers/interface/ForwardDetLayer.h"
+#include "TrackingTools/GeomPropagators/interface/Propagator.h"
+#include "TrackingTools/KalmanUpdators/interface/Chi2MeasurementEstimator.h"
+#include "TrackingTools/PatternTools/interface/TSCBLBuilderWithPropagator.h"
+#include "TrackingTools/PatternTools/interface/TrajTrackAssociation.h"
+#include "TrackingTools/PatternTools/interface/Trajectory.h"
+#include "TrackingTools/Records/interface/TrackingComponentsRecord.h"
+#include "TrackingTools/Records/interface/TransientRecHitRecord.h"
+#include "TrackingTools/Records/interface/TransientTrackRecord.h"
+#include "TrackingTools/TrackRefitter/interface/TrackTransformer.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrack.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
+#include "TrackingTools/TransientTrackingRecHit/interface/TransientTrackingRecHit.h"
 
 using namespace std;
 using namespace edm;
@@ -1058,7 +1042,7 @@ void TrackExtenderWithMTDT<TrackCollection>::produce(edm::Event& ev, const edm::
     ++itrack;
   }
 
-  auto outTrksHandle = ev.put(std::move(output));
+  ev.put(std::move(output));
   ev.put(std::move(extras));
   ev.put(std::move(outhits));
 

--- a/RecoParticleFlow/PFSimProducer/plugins/SimPFProducer.cc
+++ b/RecoParticleFlow/PFSimProducer/plugins/SimPFProducer.cc
@@ -292,7 +292,7 @@ void SimPFProducer::produce(edm::StreamID, edm::Event& evt, const edm::EventSetu
   }
 
   auto blocksHandle = evt.put(std::move(blocks));
-  auto superClustersHandle = evt.put(std::move(superclusters), "perfect");
+  evt.put(std::move(superclusters), "perfect");
 
   // list tracks so we can mark them as used and/or fight over them
   std::vector<bool> usedTrack(TrackCollection.size(), false),

--- a/RecoTauTag/RecoTau/plugins/PFRecoTauTagInfoProducer.cc
+++ b/RecoTauTag/RecoTau/plugins/PFRecoTauTagInfoProducer.cc
@@ -5,30 +5,25 @@
  * authors: Ludovic Houchu
  */
 
+#include <memory>
+
+#include <CLHEP/Random/RandGauss.h>
+#include <Math/GenVector/VectorUtil.h>
+
 #include "DataFormats/JetReco/interface/JetTracksAssociation.h"
 #include "DataFormats/TauReco/interface/PFTauTagInfo.h"
 #include "DataFormats/VertexReco/interface/Vertex.h"
 #include "DataFormats/VertexReco/interface/VertexFwd.h"
-
-#include "RecoTauTag/RecoTau/interface/PFRecoTauTagInfoAlgorithm.h"
-
-#include "FWCore/Framework/interface/EventSetup.h"
 #include "FWCore/Framework/interface/Event.h"
-#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/EventSetup.h"
 #include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
 #include "FWCore/Framework/interface/global/EDProducer.h"
-
-#include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
-
-#include <FWCore/ParameterSet/interface/ConfigurationDescriptions.h>
-#include <FWCore/ParameterSet/interface/ParameterSetDescription.h>
-
-#include "CLHEP/Random/RandGauss.h"
-
-#include "Math/GenVector/VectorUtil.h"
-
-#include <memory>
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "RecoTauTag/RecoTau/interface/PFRecoTauTagInfoAlgorithm.h"
 
 using namespace reco;
 using namespace edm;
@@ -108,7 +103,6 @@ void PFRecoTauTagInfoProducer::produce(edm::StreamID, edm::Event& iEvent, const 
     resultExt->push_back(myPFTauTagInfo);
   }
 
-  //  OrphanHandle<PFTauTagInfoCollection> myPFTauTagInfoCollection=iEvent.put(std::move(resultExt));
   iEvent.put(std::move(resultExt));
 }
 

--- a/RecoTracker/SingleTrackPattern/src/CosmicTrackFinder.cc
+++ b/RecoTracker/SingleTrackPattern/src/CosmicTrackFinder.cc
@@ -4,24 +4,23 @@
 #include <memory>
 #include <string>
 
-#include "RecoTracker/SingleTrackPattern/interface/CosmicTrackFinder.h"
-#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackExtra.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
 #include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHitCollection.h"
 #include "DataFormats/TrackerRecHit2D/interface/SiStripMatchedRecHit2DCollection.h"
 #include "DataFormats/TrackerRecHit2D/interface/SiStripRecHit2DCollection.h"
-#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
 #include "FWCore/Framework/interface/ESHandle.h"
 #include "FWCore/Framework/interface/EventSetup.h"
-#include "DataFormats/TrackReco/interface/Track.h"
-#include "DataFormats/TrackReco/interface/TrackFwd.h"
-#include "DataFormats/TrackReco/interface/TrackExtra.h"
-#include "TrackingTools/PatternTools/interface/Trajectory.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
-#include "TrackingTools/PatternTools/interface/TSCPBuilderNoMaterial.h"
 #include "FWCore/Utilities/interface/InputTag.h"
-#include "TrackingTools/PatternTools/interface/TrajTrackAssociation.h"
-
+#include "RecoTracker/SingleTrackPattern/interface/CosmicTrackFinder.h"
 #include "RecoTracker/TransientTrackingRecHit/interface/Traj2TrackHits.h"
+#include "TrackingTools/PatternTools/interface/TSCPBuilderNoMaterial.h"
+#include "TrackingTools/PatternTools/interface/TrajTrackAssociation.h"
+#include "TrackingTools/PatternTools/interface/Trajectory.h"
 
 namespace cms {
 
@@ -163,7 +162,7 @@ namespace cms {
         // protection againt invalid initial states
         if (!firstState.isValid()) {
           edm::LogWarning("CosmicTrackFinder") << "invalid innerState, will not make TrackCandidate";
-          edm::OrphanHandle<TrackCandidateCollection> rTrackCand = e.put(std::move(output));
+          e.put(std::move(output));
           return;
         }
 
@@ -171,7 +170,7 @@ namespace cms {
         if (firstId != recHits.front().rawId()) {
           edm::LogWarning("CosmicTrackFinder") << "Mismatch in DetID of first hit: firstID= " << firstId
                                                << "   DetId= " << recHits.front().geographicalId().rawId();
-          edm::OrphanHandle<TrackCandidateCollection> rTrackCand = e.put(std::move(output));
+          e.put(std::move(output));
           return;
         }
 
@@ -180,6 +179,7 @@ namespace cms {
         output->push_back(TrackCandidate(recHits, theTraj.seed(), state, theTraj.seedRef()));
       }
     }
-    edm::OrphanHandle<TrackCandidateCollection> rTrackCand = e.put(std::move(output));
+    e.put(std::move(output));
   }
+
 }  // namespace cms

